MentorMatchmaker/MentorMatchmaker/MatchMakingDomainOperations.fs

```diff
@@ -248,12 +293,62 @@ module Matchmaking =
         let unmatchedApplicants = transformedMenteesInUnmatchedApplicants @ transformedMentorsInUnmatchedApplicants
 
         let fileContent = 
             unmatchedApplicants 
             |> List.map(fun application -> $"{dumpToFileApplicationData application}")
             |> String.concat("\n")
 
-        System.IO.File.WriteAllText("applicationDataDump.txt", fileContent)
+        System.IO.File.WriteAllText("applicationDataDumpUnmatchedApplications.txt", fileContent)
+
+        (* Dump unmatched applications -- Pairing permutations *)
+        
+        let unmatchedApplicantsPairings = createUnmatchedApplicantPairingPermutations transformedMenteesInUnmatchedApplicants transformedMentorsInUnmatchedApplicants
+
+        let renderMentorPairingPermutations (permutations: {| Mentees: {| Mentee: UnmatchedApplicant; Mentor: UnmatchedApplicant; SessionHours: OverlapSchedule list |} list; Mentor: UnmatchedApplicant |}) =
+            let renderTopics applicant = 
+                let interests = applicant.Interests |> NonEmptyList.map(fun x -> x.Category.CategoryName) |> String.concat(", ")
+                $"Topics: {interests}\n"
+
+            let dumpMeetingTimes (meetingTimes: OverlapSchedule nel) =
+                meetingTimes
+                |> NonEmptyList.map(fun meetingDay ->
+                    let aggregatedTimes = meetingDay.MatchedAvailablePeriods |> NonEmptyList.toList |> List.fold(fun accumulatedTimes currentTime -> accumulatedTimes + $", {currentTime.UtcStartTime}") ""
+                    let aggregatedTimes = aggregatedTimes.Substring(2)
+                    $"{meetingDay.Weekday}: {aggregatedTimes}"
+                )
+                |> String.concat " ;"
+                
+            let renderMentee (mentee: UnmatchedApplicant) ((firstHour :: otherHours): OverlapSchedule list) =
+                let menteeData = $"\n    Mentee: {mentee.Name} ({mentee.EmailAddress})\n"
+                let menteeInterests = "    " + renderTopics mentee
+                let sessionHours = "        " + dumpMeetingTimes (NonEmptyList.create firstHour otherHours)
+
+                $"\n{menteeData}{menteeInterests}{sessionHours}"
+
+            let divider = "\n------------------------------------------------\n\n\n"
+            let header = $"Mentor: {permutations.Mentor.Name} ({permutations.Mentor.EmailAddress})\n"
+
+            let mentorTopics = renderTopics permutations.Mentor
+            
+            let menteeData = permutations.Mentees |> List.map (fun x -> renderMentee x.Mentee x.SessionHours) |> String.concat ", "
+
+            $"{divider}{header}{mentorTopics}{menteeData}"
+                
+
+
+        let pairingPermutationsFileContent =
+            unmatchedApplicantsPairings
+            |> List.map renderMentorPairingPermutations
+            |> String.concat ("\n")
+
+        
+        System.IO.File.WriteAllText("applicationDataDumpUnmatchedApplicationsPairingPermutations.txt", pairingPermutationsFileContent)
```

Comment:
I think we could pipe this into the write all text, no?

Reply:
It's been a while but I'm pretty sure WriteAllText is a member (and in that case doesn't accept a tuple).

Reply:
It's a matter of preference I guess. Both are fine for me, so I'll change it to a pipe.